integrations/artifactory.rst

```diff
@@ -0,0 +1,69 @@
+.. _integrations_artifactory:
+
+Artifactory
```

Comment:
I am not sure is this is the right place, but I want to have all the the "high level" docs about what it is and the various features in one page so there's less duplication

I would find this helpful trying to share with my boss, "here's why we could pay for this" and only have one page.